### PR TITLE
Watson Assistant + Watson Discovery + OpenAI starter kit

### PR DESCRIPTION
**Pull request by mdswift**
Add actions.json file, instructions, and example to configure Watson assistant to:
- search a project in Watson Discovery
- configure an OpenAI model prompt with the query and the Discovery search results
- generate an answer from the OpenAI model

### CODE REVIEW

File: integrations/extensions/starter-kits/language-model-conversational-search/README.md [hunk 1]
```diff
@@ -1,8 +1,12 @@
 # Language Model Conversational Search starter kit
 
-You can use this starter kit to configure language models with Watson Assistant for conversational search. This starter kit shows two ways to do this. The first is an example of [how the IBM watsonx tech preview language model and Watson Discovery search can be used to provide a personalized response to a customer](#example-1-connect-your-assistant-to-a-watsonx-language-model-via-custom-extensions). A second example shows how to use [semantic search output as input to OpenAI](#example-2-connect-your-assistant-to-hugging-face-milvus-and-openai-via-custom-extensions).
+This starter kit has multiple examples of how to configure language models with Watson Assistant for conversational search.
 
-The [prerequisites](#prerequisites) are the same for both examples.
```
Comment (jwm4) [inline integrations/extensions/starter-kits/language-model-conversational-search/README.md:5]: "both" should be "all" now

File: integrations/extensions/starter-kits/language-model-conversational-search/README.md [hunk 2]
```diff
@@ -1,8 +1,12 @@
...
+1. The first is an example of [how the IBM watsonx tech preview language model and Watson Discovery search can be used to provide a personalized response to a customer](#example-1-connect-your-assistant-to-a-watsonx-language-model-via-custom-extensions)
+1. The second example shows how to use [semantic search output as input to OpenAI](#example-2-connect-your-assistant-to-hugging-face-milvus-and-openai-via-custom-extensions)
+1. The third example shows how to use [Watson Discovery for search and use those results as input for OpenAI](#example-3-connect-your-assistant-to-watson-discovery-and-openai-via-custom-extensions)
```
Comment (jwm4) [inline integrations/extensions/starter-kits/language-model-conversational-search/README.md:7]: We should probably tweak these three bullets to use the same structure so it is easier to spot the differences.  For example, right now, the first bullet lists the model and then the search while the second two both list the search and then the model.

**Review by jwm4 — CHANGES_REQUESTED** [review]
I made some very small comments about the intro text.  The new section looks good as is.  I like the use of links to earlier sections to avoid rewriting the same stuff.

File: integrations/extensions/starter-kits/language-model-conversational-search/README.md (hunk 1 above)
Comment (mdswift) [reply integrations/extensions/starter-kits/language-model-conversational-search/README.md:5]: Agree. This is the old text, it has already been replaced.

File: integrations/extensions/starter-kits/language-model-conversational-search/README.md (hunk 2 above)
Comment (mdswift) [reply integrations/extensions/starter-kits/language-model-conversational-search/README.md]: Bullet points revised to use the same structure for clarity.

File: integrations/extensions/starter-kits/language-model-conversational-search/README.md [hunk 3]
```diff
@@ -177,8 +181,70 @@ If you want to experiment with different prompts, do the following:
...
+
+**NOTE**: If you import the actions _before_ configuring the extensions, you will see some errors on the actions because it could not find the extensions. Simply configure the extensions as described above and re-import the action JSON file.
+
+#### Session variables
+
+These are the session variables used in this example. Several of them are automatically set in the sample [discovery-openai-actions.json](discovery-openai-actions.json). You will need to set the `discovery_project_id` to point to the project id for your Watson Discovery collection.
+
+- `discovery_date_version` - Discovery date versions are documented in the [release notes](https://cloud.ibm.com/docs/discovery-data?topic=discovery-data-release-notes).
+- `discovery_project_id`: You **MUST** set this value to be the project ID obtained [when configuring Watson Discovery above](#configure-watson-discovery-extension).
+- `messages` : Input to the OpenAI model; includes the `search_results` and the `model_prompt`.
```
Comment (anirudhaBapat) [inline integrations/extensions/starter-kits/language-model-conversational-search/README.md:232]: This is a bit confusing. Can you add some example here?

Comment (anirudhaBapat) [reply integrations/extensions/starter-kits/language-model-conversational-search/README.md:232]: Should we set the initial value here for that in-session variable?

File: integrations/extensions/starter-kits/language-model-conversational-search/README.md [hunk 4]
```diff
@@ -177,8 +181,70 @@ If you want to experiment with different prompts, do the following:
...
+- `model_for_chat` : The OpenAI model used, defaults to `gpt-3.5-turbo`.
+- `model_max_tokens` : The maximum number of text fragments to input to the model. The starter kit uses 200.
+- `model_prompt`: You MAY change this to do prompt engineering, but a default will be used by the model if you don’t pass a prompt here.
+- `model_response`: The text generated by the model in response to the `messages`.
+- `passages` : Concatenation of top search results.
+- `query_text`: You MAY change this to pass queries to Watson Discovery. By default the Search action passes the user’s input.text directly.
+- `search_results`: Response object from [Discovery query](https://cloud.ibm.com/apidocs/discovery-data#query).
+- `snippet` : Top results from the Watson Discovery document search.
```
Comment (anirudhaBapat) [inline integrations/extensions/starter-kits/language-model-conversational-search/README.md:240]: I saw that you have mentioned one required param which is project_id, but specifying that these session variables need not be set and are optional can reduce some confusion.

File: integrations/extensions/starter-kits/language-model-conversational-search/README.md [hunk 5]
```diff
@@ -177,8 +181,70 @@ If you want to experiment with different prompts, do the following:
...
+
+1. Use **Actions Global Settings** to upload the JSON file to your assistant. For more information, see [Uploading](https://cloud.ibm.com/docs/watson-assistant?topic=watson-assistant-admin-backup-restore#backup-restore-import).
+
+1. Under "Variables"/"Created by you" (within the Actions page), set the `discovery_project_id` session variable using the project ID value you obtained [when configuring Watson Discovery above](#configure-watson-discovery-extension).
+
+**NOTE**: If you import the actions _before_ configuring the extensions, you will see some errors on the actions because it could not find the extensions. Simply configure the extensions as described above and re-import the action JSON file.
+
+#### Session variables
+
+These are the session variables used in this example. Several of them are automatically set in the sample [discovery-openai-actions.json](discovery-openai-actions.json). You will need to set the `discovery_project_id` to point to the project id for your Watson Discovery collection.
```
[the remark this answers is not in the document]
Comment (mdswift) [reply integrations/extensions/starter-kits/language-model-conversational-search/README.md:228]: No, that shouldn't be there. I will upload a new actions file without it.

File: integrations/extensions/starter-kits/language-model-conversational-search/README.md (hunk 3 above)
Comment (mdswift) [reply integrations/extensions/starter-kits/language-model-conversational-search/README.md:232]: I am not sure what you mean by this `Should we set the initial value here for that in-session variable?`. Could you please clarify?

**Review by anirudhaBapat — APPROVED** [review]
Actually, I needed to set Discovery using basic auth (and not bearer one which is used in cp4d)

Screenshot 2023-06-27 at 9 37 42 AM

And that worked!!
Screenshot 2023-06-27 at 9 39 36 AM

Just had few minor comments in Readme, otherwise Looking awesome!

File: integrations/extensions/starter-kits/language-model-conversational-search/README.md (hunk 3 above)
Comment (anirudhaBapat) [reply integrations/extensions/starter-kits/language-model-conversational-search/README.md:232]: Oh I mean in the session variable like how we set project_id

File: integrations/extensions/starter-kits/language-model-conversational-search/README.md (hunk 4 above)
Comment (mdswift) [reply integrations/extensions/starter-kits/language-model-conversational-search/README.md:240]: The README says that most of these variables are automatically set in the actions file. That is not the same as saying that they are optional or don't need to be set. I will clarify that text.

**Review by jwm4 — APPROVED** [review]

**Pull request merged**